src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -360,6 +363,14 @@ public void Dispose()
                 }
             }
 
+            public void Cancel()
+            {
+                lock (SyncObject)
+                {
+                    Task ignored = _connection.SendRstStreamAsync(_streamId, Http2ProtocolErrorCode.Cancel);
+                }
```

Comment:
What about _streamState here?

Reply:
I've gone ahead and updated this, but I'm still trying to wrap my head around the other changes that came in with the same PR. Will I need to update this to follow the same pattern as OnResponseAbort where I need to signal the next waiter?

Reply:
I put some thoughts on this and related issues in #35652. Thoughts on that?

Reply:
I've added some thoughts on the linked issue. I think we should probably add some additional values to StreamState (i.e, Canceled) when we make the changes described there but for now I've set it as Aborted.